src/lib/components/FileUpload/FileUpload.tsx

```diff
@@ -1,23 +1,167 @@
-import { useDropzone, DropzoneOptions } from "react-dropzone";
+import { ReactNode, useCallback, useId, useState } from "react";
+
+import { Button, Icon, Label } from "@canonical/react-components";
+import classNames from "classnames";
+import { useDropzone, DropzoneOptions, FileRejection } from "react-dropzone";
+import "./FileUpload.scss";
 
 export interface FileUploadProps {
+  error?: ReactNode;
+  files: File[];
+  help?: string;
+  label?: string;
+  maxFiles?: number;
+  maxSize?: number;
   onFileUpload: NonNullable<DropzoneOptions["onDrop"]>;
+  rejectedFiles: FileRejection[];
+  removeFile: (file: File) => void;
+  removeRejectedFile: (fileRejection: FileRejection) => void;
 }
 
-import "./FileUpload.scss";
-
 export const FileUpload: React.FC<FileUploadProps> = ({
+  error,
+  files,
```

Comment:
```suggestion
  acceptedFiles,
```
A file can only be either accepted or rejected. Having said that, I think we could streamline state of this component, which I describe in detail below.